pennylane/ops/op_math/condition.py

```diff
@@ -430,3 +446,127 @@ def wrapper(*args, **kwargs):
+
+    cond_prim = jax.core.Primitive("cond")
+    cond_prim.multiple_results = True
+
+    @cond_prim.def_impl
+    def _(conditions, *args_and_consts, jaxpr_branches, n_consts_per_branch, n_args):
+
+        args = args_and_consts[:n_args]
+        consts_flat = args_and_consts[n_args:]
+
+        consts_per_branch = []
+        start = 0
+        for n in n_consts_per_branch:
+            consts_per_branch.append(consts_flat[start : start + n])
+            start += n
+
+        for pred, jaxpr, consts in zip(conditions, jaxpr_branches, consts_per_branch):
+            if pred and jaxpr is not None:
+                return jax.core.eval_jaxpr(jaxpr.jaxpr, consts, *args)
+
+        return ()
+
+    @cond_prim.def_abstract_eval
+    def _(*_, jaxpr_branches, **__):
+
+        outvals_true = jaxpr_branches[0].out_avals
+
+        for idx, jaxpr_branch in enumerate(jaxpr_branches):
+            if idx == 0:
+                continue
+
+            if jaxpr_branch is None:
+                if outvals_true:
+                    raise ValueError(
+                        "The false branch must be provided if the true branch returns any variables"
+                    )
+                # this is tested, but coverage does not pick it up
+                continue  # pragma: no cover
+
+            outvals_branch = jaxpr_branch.out_avals
+            branch_type = "elif" if idx < len(jaxpr_branches) - 1 else "false"
+            _validate_abstract_values(outvals_branch, outvals_true, branch_type, idx - 1)
+
+        # We return the abstract values of the true branch since the abstract values
+        # of the other branches (if they exist) should be the same
+        return outvals_true
+
+    return cond_prim
+
+
+def _capture_cond(condition, true_fn, false_fn=None, elifs=()) -> Callable:
+    """Capture compatible way to apply conditionals."""
+
+    import jax  # pylint: disable=import-outside-toplevel
+
+    cond_prim = _get_cond_qfunc_prim()
+
+    elifs = (elifs,) if len(elifs) > 0 and not isinstance(elifs[0], tuple) else elifs
+
+    @wraps(true_fn)
+    def new_wrapper(*args, **kwargs):
+
+        jaxpr_true = jax.make_jaxpr(functools.partial(true_fn, **kwargs))(*args)
+        jaxpr_false = (
+            jax.make_jaxpr(functools.partial(false_fn, **kwargs))(*args) if false_fn else None
+        )
+
+        # We extract each condition (or predicate) from the elifs argument list
+        # since these are traced by JAX and are passed as positional arguments to the primitive
+        elifs_conditions = []
+        jaxpr_elifs = []
+
+        for pred, elif_fn in elifs:
+            elifs_conditions.append(pred)
+            jaxpr_elifs.append(jax.make_jaxpr(functools.partial(elif_fn, **kwargs))(*args))
+
+        conditions = jax.numpy.array([condition, *elifs_conditions, True])
+
+        jaxpr_branches = [jaxpr_true, *jaxpr_elifs, jaxpr_false]
+        jaxpr_consts = [jaxpr.consts if jaxpr is not None else () for jaxpr in jaxpr_branches]

```

Comment:
```suggestion
        jaxpr_consts = (jaxpr.consts if jaxpr is not None else () for jaxpr in jaxpr_branches)
```
since we will only be using this to iterate over a single time, we can use a generator.